test/Microsoft.AspNetCore.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs

```diff
@@ -865,7 +941,22 @@ public void ModelStateDictionary_HasReachedMaxErrors(int errorCount, bool expect
         }
 
         [Fact]
-        public void ModelStateDictionary_ReturnGenericErrorMessage_WhenModelStateNotSet()
+        public void ModelStateDictionary_ReturnExceptionMessage_WhenModelStateNotSet()
+        {
+            // Arrange
+            var expected = "The supplied value is invalid for Length.";
+            var dictionary = new ModelStateDictionary();
+
+            // Act
+            dictionary.TryAddModelError("key", new FormatException("The supplied value is invalid for Length."));
```

Comment:
Move the `FormatException` into `// Arrange` and use it in an `Assert.Same(exception, error.Exception);` instead of the last line